src/target/riscv32.c

```diff
@@ -141,6 +150,45 @@ static void riscv32_regs_write(target_s *const target, const void *const data)
 	riscv_csr_write(hart, RV_DPC, &regs->pc);
 }
 
+static ssize_t riscv32_bool_to_4(bool ret)
```

Comment:
Please mark this `inline` as in `static inline ssize_t`, the body of the function is so short that we'll likely get better codegen to have the compiler inline this.